### DISCUSSION TITLE
Improvement: Artisan make:enum Must Follow Consistent Directory Creation Behavior

### DISCUSSION THREAD

**Discussion by usmanx03** [laravel/framework#53674] [Ideas] [answered]
When generating enums using Artisan, they are currently placed in the root directory if the App/Enums directory does not exist. If the App/Enums directory is present, the generated enums are correctly placed within it.

However, this behavior differs from other make commands, which follow a consistent standard for directory creation. To maintain uniformity across all Artisan commands, the EnumMakeCommand should adhere to the same approach, creating the App/Enums directory if it doesn’t already exist.

Relevant Command Implementation:
[EnumMakeCommand Source](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Foundation/Console/EnumMakeCommand.php)

**Comment by huzaifa1-0** [comment]
This is a practical suggestion. Other Artisan commands, like make:model, create necessary directories automatically. Aligning `php artisan make:enum` with this behavior by creating the App/Enums directory when missing ensures consistency and avoids manual setup.

**Reply by dcgitx** [reply, marked as answer]
At the moment I see this is allowing for folks who 'may' use a directory named 'Enumerations' instead of the shorthand 'Enums' this is probably why it is like it is.. make:job/listener etc it's a simple word that it can be assumed everyone will use...

**Reply by usmanx03** [reply]
I guess the same goes for the Interfaces/Contracts directory and make command. Ok then.
